### PR TITLE
mgr: add throttle policy for DaemonServer

### PR DESCRIPTION
**Pull request by ericqzhao**
This commit fix the throttle parameter of osd does not take effect for mgr Fixes: https://tracker.ceph.com/issues/61942

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/mgr/DaemonServer.cc [hunk 1]
```diff
@@ -123,7 +123,16 @@ int DaemonServer::init(uint64_t gid, entity_addrvec_t client_addrs)
...
-
+  // throttle policy
+  msgr->set_policy(entity_name_t::TYPE_OSD,
+                   Messenger::Policy::stateless_server(
+                     CEPH_FEATURE_SERVER_LUMINOUS));
+  msgr->set_policy(entity_name_t::TYPE_MON,
+                   Messenger::Policy::lossy_client(CEPH_FEATURE_UID |
+                                                   CEPH_FEATURE_PGID64));
+  msgr->set_policy(entity_name_t::TYPE_MDS,
+                   Messenger::Policy::lossless_peer(CEPH_FEATURE_UID));
```
Comment (batrick) [inline src/mgr/DaemonServer.cc:134]:
```suggestion
  msgr->set_policy(entity_name_t::TYPE_MDS,
                    Messenger::Policy::stateless_server(
                     CEPH_FEATURE_SERVER_LUMINOUS));
```

I believe this is correct. For this messenger instance, the mgr should only be receiving perf metrics from the MDS.

Comment (ericqzhao) [reply src/mgr/DaemonServer.cc]: You're right

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
Err, please squash my suggested changes into your first commit.

**Comment by ericqzhao** [comment]
> Err, please squash my suggested changes into your first commit.

done

**Comment by ericqzhao** [comment]
> Err, please squash my suggested changes into your first commit.
Need-qa ?What do you need me to do?

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcJxeRJYse1822-wip-yuri4-testing-2023-08-10-1739

**Comment by ljflores** [comment]
jenkins retest this please

**Comment by ljflores** [comment]
Some checks failed so I retriggered them. Please merge at will.

**Pull request merged**